### PR TITLE
Collection | Enhancement | Default type and template for View, part 2

### PR DESCRIPTION
**Pull request by konstantiniiv**

### CODE REVIEW

File: app/src/main/java/com/anytypeio/anytype/ui/sets/ObjectSetFragment.kt [hunk 1]
```diff
@@ -1217,6 +1227,12 @@ open class ObjectSetFragment :
         inflater, container, false
     )
 
+    override fun onProceedWithUpdateType(id: Id) {
+        vm.onNewTypeForViewerClicked(id)
+    }
+
+    override fun onProceedWithDraftUpdateType(id: Id) {}
```
Comment (uburoiubu) [inline app/src/main/java/com/anytypeio/anytype/ui/sets/ObjectSetFragment.kt:1234]:
```
// Do nothing
```

File: domain/src/main/java/com/anytypeio/anytype/domain/dataview/interactor/CreateDataViewObject.kt [hunk 2]
```diff
@@ -168,7 +171,8 @@ class CreateDataViewObject(
 
     data class Result(
         val objectId : Id,
-        val objectType: Id?
+        val objectType: Id?,
+        val struct: Struct?
```
Comment (uburoiubu) [inline domain/src/main/java/com/anytypeio/anytype/domain/dataview/interactor/CreateDataViewObject.kt:175]: do we need nullable here? we could as well return an empty map

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ObjectSetViewModel.kt [hunk 3]
```diff
@@ -1698,12 +1711,13 @@ class ObjectSetViewModel(
                         Pair(selectedTypeId, types)
                     }
                 }.map { (selectedTypeId, types) ->
-                    types.map { type ->
-                        TemplateObjectTypeView.Item(
-                            type = ObjectWrapper.Type(type.map),
-                            isDefault = type.id == selectedTypeId.id
-                        )
-                    }
+                    listOf(TemplateObjectTypeView.Search) +
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ObjectSetViewModel.kt:1714]:
```
buildList {
     add(TemplateObjectTypeView.Search)
     addAll(types.map {....})
}
```

**Review by uburoiubu — APPROVED** [review]

File: app/src/main/java/com/anytypeio/anytype/ui/sets/ObjectSetFragment.kt (hunk 1 above)
Comment (konstantiniiv) [reply app/src/main/java/com/anytypeio/anytype/ui/sets/ObjectSetFragment.kt:1234]: done

File: domain/src/main/java/com/anytypeio/anytype/domain/dataview/interactor/CreateDataViewObject.kt (hunk 2 above)
Comment (konstantiniiv) [reply domain/src/main/java/com/anytypeio/anytype/domain/dataview/interactor/CreateDataViewObject.kt:175]: changed to empty map

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ObjectSetViewModel.kt (hunk 3 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ObjectSetViewModel.kt]: fixed

**Pull request merged**